rest-server/core/src/main/scala/com/actionml/core/search/SearchSupport.scala

```diff
@@ -92,4 +92,5 @@ trait SearchClient[R, D] {
   def search(query: SearchQuery): Seq[R]
   def searchAsync(query: SearchQuery): Future[Seq[R]]
   def findDocById(id: String): D
+  def findDocByIdAsync(id: String): Future[D]
```

Comment:
@andreyshikov This makes me wonder if we should make create and delete index async since they could take a long time indeed. This is analogous to what you did to make changing the TTL async